src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -2178,6 +2176,42 @@ FixedwingPositionControl::control_manual_position(const float control_interval,
 	_att_sp.pitch_body = get_tecs_pitch();
 }
 
+void FixedwingPositionControl::control_approach_transition(const float control_interval, const Vector2f &ground_speed,
+		const position_setpoint_s &pos_sp_prev,
+		const position_setpoint_s &pos_sp_curr)
+{
+	float target_airspeed = adapt_airspeed_setpoint(control_interval, pos_sp_curr.cruising_speed,
+				_performance_model.getMinimumCalibratedAirspeed(getLoadFactor()), ground_speed);
+
+	Vector2f curr_pos_local{_local_pos.x, _local_pos.y};
+	Vector2f curr_wp_local = _global_local_proj_ref.project(pos_sp_curr.lat, pos_sp_curr.lon);
+
+	_npfg.setAirspeedNom(target_airspeed * _eas2tas);
+	_npfg.setAirspeedMax(_performance_model.getMaximumCalibratedAirspeed() * _eas2tas);
+
+	if (_position_setpoint_previous_valid && pos_sp_prev.type != position_setpoint_s::SETPOINT_TYPE_TAKEOFF) {
+		Vector2f prev_wp_local = _global_local_proj_ref.project(pos_sp_prev.lat, pos_sp_prev.lon);
+		navigateLine(prev_wp_local, curr_wp_local, curr_pos_local, ground_speed, _wind_vel);
+
+	} else {
+		navigateWaypoint(curr_wp_local, curr_pos_local, ground_speed, _wind_vel);
```

Comment:
With this, it can happen again, that we make a hard turn after we past the waypoit, no? Shouldn't we make it like that: if the current setpoint is a loiter, then navigateLoiter, if it is velocity, then navigatePathTangent, if it is position and has previous navigate line previous to current, and else navigateLine vehicle position to current setpoint position?
I am not sure if this is complete, but i think you want to avoid that mif transitioning it willreverse the direction by passing by.

Reply:
@KonradRudin Yes, in theory it can. I have verified that it should under normal circumstances not happen when you plan a mission landing or when you use the home land approaches. I am currently actually testing what happens when you use a plan transition command.
I am not opposed to add logic (similar to what we have done on the older releases) which handles that case.